src/hist.jl

```diff
@@ -14,6 +16,31 @@ function _check_closed_arg(closed::Symbol, funcsym)
 end
 
 
+## Fast map for tuples (a lot faster than using Base.map)
+@generated function _tuple_map{N}(f, tuples::NTuple{N, Any}...)
```

Comment:
I checked - broadcast is equally slow (I wrote a little tuple-map benchmark, independent of StatsBase). I guess you're right tough - I was thinking about the fact that a PR on this for `Base` wouldn't make it into 0.6 anymore, so ... but maybe it's really best not to put a temporary shim in here. I'll take it out.